### ISSUE TITLE
Adding a new native (sidechain) compressor

### ISSUE THREAD

**Issue by PhysSong** [LMMS/lmms#3817]
LMMS currently have various native effects such as EQ, Reverb, Delay, and so on. However, There are no native compressors/limiters in LMMS.
And sidechaining is not easy, since it requires peak controllers(and external sidechain compressors for better quality of output), so it can't be done with just one effect in effect chain. If a native plugin supports it, it would be much easier than now.
Implementing a compressor isn't too hard, but sidechaining will require some major changes in our codebase. It requires audio signals from FX channels and/or individual tracks, which is a functionality not in LMMS yet.
So any ideas about this?

**Comment by Umcaruje** [comment]
we could implement an audio send plugin/controller that would handle sending the sidechain signal to the compressor.

I really agree with the idea of a native compressor, because it would allow for visual feedback when compressing, I find this very crucial when using other software

**Comment by tresf** [comment]
> visual feedback [...] crucial

Agreed.

> could implement an audio send plugin/controller that would handle sending the sidechain signal to the compressor.

It's going to get tricky to handle the various scenarios that the mixer can already do.  Perhaps for starters we just make a wizard that assumes 1 channel compresses 1 other channels and links the peak controller to the effect wet/dry (**cough** #3262, **cough**).  Side-chaining -- although common for mixing can really be exploited for so many other purposes, but reinventing what the mixer already does well should be carefully avoided.

**Comment by musikBear** [comment]
what are your opinion on the way FLs handles 'sidechaining' ?
They draw a graph for the volume-output in the whole bar
It opens for any creative form of output and are not at all related to what happens in other mixer-channels. -Or is that not proper sidechaining at all? 
Beside that- 👍 for a native compressor

**Comment by tresf** [comment]
> It opens for any creative form of output and are not at all related to what happens in other mixer-channels. 

Right, this is less of a side-chaining display and more of an "effective effects filter" display.  We really need a spectrum analyser for all effects.  This is just a place where it becomes pressing because the real-time display is needed for adjusting levels.

I currently do this with an EQ now, but it only shows the after, not the before vs. the after, like most compressor UIs would.

**Comment by PhysSong** [comment]
#2993 seems to be related, but a little bit different. Could we merge two issues into single issue?

**Comment by LostRobotMusic** [comment]
LMMS now has a native multiband compressor with #6925.

**Comment by M4rotte** [comment]
>LMMS now has a native multiband compressor with #6925.

LMMS still has some laughable aspects from the point of view of many DAW users, but IMO,  that for the last couple of years, it made some crazy progress.

I learnt the existence of the native compressor far too late, I feel fine to know, just right now, that there is multiband one too!

**Comment by tresf** [comment]
> LMMS still has some laughable aspects

Hi,

Let's please be kind to the few volunteers that dedicate their time to work on the project.  Calling the software "laughable" is strictly forbidden here as it has the tendency to being objective topics like this one off the rails.

**Comment by musikBear** [comment]
By chance i was looking at LMMS' compressor the other day, and i was disturbed by the way the controls obstructed the monitoring window. My 1.3 is not brand new i have to admit, but is that obstructing pr design?
![image](https://github.com/LMMS/lmms/assets/6662066/16b201c0-3bf8-4bbd-a4ce-54c4650e4142)

**Comment by LostRobotMusic** [comment]
@musikBear I agree it covers far more than it should and I'm planning a redesign eventually.  Do note that the plugin is resizable, and you can also change the volume scale by scrolling in the background, both of which can help get it out of the way of the important info for now.
